control/controlhttp/client.go

```diff
@@ -382,18 +383,27 @@ func (a *Dialer) dialURL(ctx context.Context, u *url.URL, addr netip.Addr) (*Cli
 func (a *Dialer) tryURLUpgrade(ctx context.Context, u *url.URL, addr netip.Addr, init []byte) (net.Conn, error) {
 	var dns *dnscache.Resolver
 
+	// Maintain prior behaviour where passing no logger to
```

Comment:
logf method instead?

Reply:
I intentionally didn't use that; the `a.logf` method will drop logs if no `a.Logf` option is specified, whereas the previous behaviour of `dnscache.Resolver` always logs with `log.Printf` if there's no logger specified.

Reply:
logfOrDefault method then? I just don't want to see a big comment and var decl/conditional in the middle of unrelated code

Reply:
I actually just changed this to use `a.Logf` directly in the argument; this propagates nil-ness and retains the default "fall back to log.Printf" behaviour that `dnscache.Resolver` has. PTAL?